src/mono/wasm/test-main.js

```diff
@@ -361,10 +361,24 @@ async function loadDotnet(file) {
             return require(file);
         };
     } else if (is_browser) { // vanila JS in browser
-        loadScript = async function (file) {
-            globalThis.exports = {}; // if we are loading cjs file
-            const createDotnetRuntime = await import(file);
-            return typeof createDotnetRuntime === "function" ? createDotnetRuntime : globalThis.exports.createDotnetRuntime;
+        loadScript = function (file) {
+            var loaded = new Promise((resolve, reject) => {
+                try {
+                    globalThis.__onDotnetRuntimeLoaded = (createDotnetRuntime) => {
+                        // this is callback we have in CJS version of the runtime
+                        resolve(createDotnetRuntime);
+                    };
+                    import(file).then(({ default: createDotnetRuntime }) => {
+                        // this would work with ES6 default export
+                        if (createDotnetRuntime) {
+                            resolve(createDotnetRuntime);
+                        }
+                    });
+                } catch (err) {
+                    reject(err);
```

Comment:
Does this try catch work with `import` and no `await`?
Or do you need a failture handler like this:

```js
import(file).then(({ default: createDotnetRuntime }) => {
    // this would work with ES6 default export
    if (createDotnetRuntime) {
        resolve(createDotnetRuntime);
    }
}, (err) => {
    reject(err);
});
```

Reply:
Thanks for the suggestion! I tested it and indeed it works and it's nicer.